### PR TITLE
#27500 Prepare ConfigurableProduct module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by slavvka** [comment]
The cause of failure is:
```
18:23:59 FPHP Fatal error:  Interface 'Magento\Catalog\Api\Data\ProductExtensionInterface' not found in /var/www/html/app/code/Magento/ConfigurableProduct/Test/Unit/Model/Product/ProductExtensionAttributes.php on line 10

```

File: app/code/Magento/ConfigurableProduct/Test/Unit/Observer/HideUnsupportedAttributeTypesTest.php [hunk 1]
```diff
@@ -44,11 +44,11 @@ public function testExecuteWhenBlockNotPassed()
     }
 
     /**
-     * @param RequestInterface|\PHPUnit_Framework_MockObject_MockObject $request
+     * @param RequestInterface|\PHPUnit\Framework\MockObject\MockObject $request
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/Test/Unit/Observer/HideUnsupportedAttributeTypesTest.php:48]: Please import it

File: app/code/Magento/ConfigurableProduct/Test/Unit/Model/Quote/Item/CartItemProcessorTest.php [hunk 2]
```diff
@@ -43,22 +57,22 @@ class CartItemProcessorTest extends \PHPUnit\Framework\TestCase
      */
     private $productOptionExtensionAttributes;
 
-    /** @var \PHPUnit_Framework_MockObject_MockObject */
+    /** @var \PHPUnit\Framework\MockObject\MockObject */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/Test/Unit/Model/Quote/Item/CartItemProcessorTest.php:62]: please import it

File: app/code/Magento/ConfigurableProduct/Test/Unit/Model/Product/SaveHandlerTest.php [hunk 3]
```diff
@@ -230,12 +228,12 @@ private function initConfigurableFactoryMock()
...
      * Mock for options save
      *
-     * @param \PHPUnit_Framework_MockObject_MockObject $attribute
+     * @param \PHPUnit\Framework\MockObject\MockObject $attribute
      * @param $sku
      * @param $id
      * @return void
      */
-    private function processSaveOptions(\PHPUnit_Framework_MockObject_MockObject $attribute, $sku, $id)
+    private function processSaveOptions(\PHPUnit\Framework\MockObject\MockObject $attribute, $sku, $id)
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/Test/Unit/Model/Product/SaveHandlerTest.php]: Please import it

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@slavvka  could you give some info why static tests failing?

**Comment by slavvka** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@ihor-sviziev 
```
19:08:55 .PHP Fatal error:  Interface 'Magento\Catalog\Api\Data\ProductExtensionInterface' not found in /var/www/html/app/code/Magento/ConfigurableProduct/Test/Unit/Model/Product/ProductExtensionAttributes.php on line 12
```

**Comment by lbajsarowicz** [comment]
@slavvka Is that message from PHPStan?

**Comment by slavvka** [comment]
@lbajsarowicz no, phpunit when running PHP static tests, only on B2B:
```
        <testsuite name="PHP Coding Standard Verification">
            <file>testsuite/Magento/Test/Php/LiveCodeTest.php</file>
        </testsuite>
        <testsuite name="Xss Unsafe Output Test">
            <file>testsuite/Magento/Test/Php/XssPhtmlTemplateTest.php</file>
        </testsuite>
```

**Comment by slavvka** [comment]
Also here are the same warnings:
```
14:59:34 There were 13 warnings:
14:59:34 
14:59:34 1) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_and_new_status" (null, null, null)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 2) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_status_and_subscribe" (null, true, true)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 3) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "new_unsubscribed_value_and_missing_previous_status" (null, false, null)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 4) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_subscribed_status_without_new_value" (1, null, null)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 5) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_subscribed_previous_and_new_status" (1, true, null)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 6) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_subscribed_customer" (1, false, false)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 7) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previously_unsubscribed_status_without_new_value" (3, null, null)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 8) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unsubscribed_customer" (3, true, true)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 9) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_unsubscribed_previous_and_new_status" (3, false, null)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 10) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_unconfirmed_status_without_new_value" (4, null, true)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 11) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unconfirmed_status" (4, true, true)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 12) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_unconfirmed_status" (4, false, true)
14:59:34 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:59:34 
14:59:34 13) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterGetByIdCreatesExtensionAttributes
14:59:34 Trying to configure method "setIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
```

**Comment by slavvka** [comment]
Static only in B2B:
```
16:48:03 .PHP Fatal error:  Interface 'Magento\Catalog\Api\Data\ProductExtensionInterface' not found in /var/www/html/app/code/Magento/ConfigurableProduct/Test/Unit/Model/Product/ProductExtensionAttributes.php on line 12
```

**Comment by slavvka** [comment]
Unit CE, EE, B2B
```
16:49:52 There were 13 warnings:
16:49:52 
16:49:52 1) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_and_new_status" (null, null, null)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 2) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_status_and_subscribe" (null, true, true)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 3) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "new_unsubscribed_value_and_missing_previous_status" (null, false, null)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 4) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_subscribed_status_without_new_value" (1, null, null)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 5) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_subscribed_previous_and_new_status" (1, true, null)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 6) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_subscribed_customer" (1, false, false)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 7) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previously_unsubscribed_status_without_new_value" (3, null, null)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 8) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unsubscribed_customer" (3, true, true)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 9) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_unsubscribed_previous_and_new_status" (3, false, null)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 10) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_unconfirmed_status_without_new_value" (4, null, true)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 11) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unconfirmed_status" (4, true, true)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 12) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_unconfirmed_status" (4, false, true)
16:49:52 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
16:49:52 
16:49:52 13) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterGetByIdCreatesExtensionAttributes
16:49:52 Trying to configure method "setIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
```

**Comment by slavvka** [comment]
```
12:36:57 PHP Fatal error:  Declaration of Magento\Banner\Test\Unit\Block\Adminhtml\Widget\Instance\Edit\Chooser\ContainerTest::setUp() must be compatible with Magento\Widget\Test\Unit\Block\Adminhtml\Widget\Instance\Edit\Chooser\AbstractContainerTest::setUp(): void in /var/www/html/app/code/Magento/Banner/Test/Unit/Block/Adminhtml/Widget/Instance/Edit/Chooser/ContainerTest.php on line 15
```

**Comment by slavvka** [comment]
Static tests
```
12:41:47 .PHP Fatal error:  Interface 'Magento\Catalog\Api\Data\ProductExtensionInterface' not found in /var/www/html/app/code/Magento/ConfigurableProduct/Test/Unit/Model/Product/ProductExtensionAttributes.php on line 12
```

**Comment by slavvka** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
From my perspective this pr could be merged, as I understood issue inside EE or b2b version

**Pull request merged**